### PR TITLE
feat: error for oneOf, anyOf, allOf schema that does not use array

### PR DESCRIPTION
**Pull request by barrett-schonefeld**
Changes:
- added default error for oneOf, anyOf, or allOf schema that is not use top-level array

Tests:
- added tests when oneOf, anyOf, or allOf schemas are nested ("oneOf allOf," for example), when oneOf, anyOf, or allOf schema is used as a property in an obect, and when oneOf, anyOf, or allOf schema used as items in array
- updated schema-ibm test to import the default config values and made necessary changes to test to account for unexpected errors and warnings resulting from having all default values present
- added test to ensure errors not issued when compose models use an array

### CODE REVIEW

File: test/plugins/validation/2and3/schema-ibm.js [hunk 1]
```diff
@@ -1005,8 +882,152 @@ describe('validation plugin - semantic - schema-ibm - OpenAPI 3', () => {
...
+    expect(res.errors[2].message).toEqual('anyOf should be an array');
+    expect(res.errors[2].path).toEqual([
+      'components',
+      'schemas',
+      'any_of_array',
+      'items',
+      'anyOf'
+    ]);
+  });
+
```
Comment (dpopp07) [inline test/plugins/validation/2and3/schema-ibm.js]: Will you add one positive test? It can just be one compose model which correct usage in one of the files

File: README.md [hunk 2]
```diff
@@ -251,6 +251,7 @@ The supported rules are described below:
 | no_property_description     | Flag any schema that contains a 'property' without a `description` field.     | shared   |
 | description_mentions_json   | Flag any schema with a 'property' description that mentions the word 'JSON'.  | shared   |
 | array_of_arrays             | Flag any schema with a 'property' of type `array` with items of type `array`. | shared   |
+| compose_model_field_not_array | Flag any compose model schema that does not use an array.                   | shared   |
```
Comment (mkistler) [inline README.md]: If we keep this configurable, I think we want a clearer name and description.  "compose model" is a term from the SDK generator -- not used in OpenAPI or JSON Schema.  So better to use something like "allOf_value_not_array" and "Flag any `allOf` schema attribute whose value is not an array".

File: src/plugins/validation/2and3/semantic-validators/schema-ibm.js [hunk 3]
```diff
@@ -50,14 +50,33 @@ module.exports.validate = function({ jsSpec, isOAS3 }, config) {
...
       current === 'items' ||
-      modelLocations.indexOf(path[path.length - 2]) > -1
+      modelLocations.includes(path[path.length - 2])
     ) {
       const pushLeafSchemas = (obj, path) => {
-        if (obj.allOf && Array.isArray(obj.allOf)) {
+        if (obj.allOf && !Array.isArray(obj.allOf)) {
+          messages.addMessage(
+            path.concat(['allOf']),
+            'allOf should be an array',
```
Comment (mkistler) [inline src/plugins/validation/2and3/semantic-validators/schema-ibm.js]:
```suggestion
            'allOf value should be an array',
```

File: test/plugins/validation/2and3/schema-ibm.js [hunk 4]
```diff
@@ -984,12 +867,6 @@ describe('validation plugin - semantic - schema-ibm - OpenAPI 3', () => {
   });
 
   it('should process allOf, oneOf, anyOf schemas correctly', () => {
-    const config = {
-      schemas: {
-        invalid_type_format_pair: 'error'
-      }
-    };
-
```
Comment (mkistler) [inline test/plugins/validation/2and3/schema-ibm.js]: All the changes above here appear to be unrelated to this feature.

Comment (barrett-schonefeld) [reply test/plugins/validation/2and3/schema-ibm.js]: Yes, why I was introducing tests to this module, I thought it would be a good idea to clean up the individual declarations of config and replace them with one import. I changed the config name to customConfig for the cases where the test uses a custom value.

File: README.md (hunk 2 above)
Comment (barrett-schonefeld) [reply README.md]: I chose to use compose model because this PR checks that `allOf`, `anyOf, and `oneOf` use arrays. In my next changes, I am going to remove the configuration option, but I want to make sure you agree with the choice to include `anyOf` and `oneOf`.

File: test/plugins/validation/2and3/schema-ibm.js (hunk 4 above)
Comment (dpopp07) [reply test/plugins/validation/2and3/schema-ibm.js]: ^ I support these changes

File: README.md (hunk 2 above)
Comment (mkistler) [reply README.md]: Yes -- we should check all three.  I made my suggestion before I saw the code below.
